### PR TITLE
test: fix failing gitsync tests

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
RCA:
Test was failing because of hard wait
Also the dropdown element which was being clicked was failing

Solution:
Removed hard waits and replaced with intercept
Updated locator and better method to fix tests

EE PR: https://github.com/appsmithorg/appsmith-ee/pull/4413

## Summary by CodeRabbit

- **Refactor**
  - Updated the method for merging changes to the master branch in Git sync tests for improved reliability.
  
- **Tests**
  - Revised Cypress test configurations and spec names for better organization and clarity.
  - Enhanced the `CheckMergeConflicts` method to improve test accuracy and reliability by including element visibility checks and network status assertions.

### CODE REVIEW

File: app/client/cypress/limited-tests.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # To run only limited tests - give the spec names in below format:
-cypress/e2e/Regression/ClientSide/Templates/Fork_Template_spec.js
+cypress/e2e/Regression/ClientSide/Git/GitSync/GitSyncedApps_spec.js
```
Comment (ApekshaBhosale) [inline app/client/cypress/limited-tests.txt:2]: revert this please!

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/limited-tests.txt (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/limited-tests.txt:2]: let me run few times and then remove and send for review, it has automatically added you as reviewer @ApekshaBhosale

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**